### ISSUE TITLE
Edit a big XML file high rocket the CPU and need to kill Emacs

### ISSUE THREAD

**Issue by RafaelGPaz** [syl20bnr/spacemacs#7794]
Hello,

After months suffering this issue, I finally decided to post the details here in seek of some help.

In my job I work with a lot with XML files. One of them, which is particularly important, is a 188Kb file which contains 2,653 lines.

Every time I edit this file, Emacs high rocket the CPU. In the best cases, if I minimise Emacs and wait for several minutes, the CPU gets back to normal. But most of the times I need to kill Emacs.

I don't have any special configuration for XML files. I tried several major modes (xml-mode, nxml-mode, web-mode) but I get the same issue with all of them. My temporary solution to edit this file is change to fundamental mode.

By the way, I'm running Emacs in a VirtualBox with Debian testing + LXDE. And  I get this issue with both,  Emacs 24 and Emacs 26.

Does anyone has this problem too?

You can download this XML file following this [Link](http://www.parallaxvirtualtours.co.uk/jobs/download/config.xml)

Try adding an new line like `<car name="foo" ....`

Alternatively, can I have some advise to investigate what is going on and try to fix it by myself?

Thanks

UPDATE:
This is what `describe-mode` returns:

Enabled minor modes: 
Anzu 
Async-Bytecomp-Package 
Auto-Composition
Auto-Compression
Auto-Encryption 
Auto-Highlight-Symbol 
Bug-Reference-Prog
Clean-Aindent
 Column-Number 
Company Counsel 
Diff-Auto-Refine 
Electric-Indent
Eval-Sexp-Fu-Flash 
Evil Evil-Commentary 
Evil-Escape 
Evil-Local
Evil-Search-Highlight-Persist 
Evil-Surround 
Eyebrowse 
File-Name-Shadow 
Flx-Ido
Flyspell 
Font-Lock 
Git-Gutter+ 
Global-Anzu 
Global-Auto-Revert 
Global-Company
Global-Eldoc 
Global-Evil-Search-Highlight-Persist 
Global-Evil-Surround
Global-Font-Lock 
Global-Git-Commit 
Global-Git-Gutter+ 
Global-Hl-Line
Global-Page-Break-Lines 
Global-Prettify-Symbols 
Global-Spacemacs-Leader-Override
Global-Spacemacs-Whitespace-Cleanup 
Global-Undo-Tree 
Global-Vi-Tilde-Fringe
Goto-Address-Prog 
Highlight-Numbers 
Highlight-Parentheses 
Hl-Todo 
Hs
Ido-Vertical 
Info-Breadcrumbs-In-Mode-Line 
Ivy 
Line-Number 
Mouse-Wheel
Override-Global 
Persp Popwin 
Rainbow-Delimiters 
Recentf 
Save-Place 
Savehist
Shell-Dirtrack 
Show-Smartparens 
Show-Smartparens-Global 
Smartparens
Smartparens-Strict 
Spaceline-Helm 
Spaceline-Info 
Spacemacs-Leader-Override
Spacemacs-Whitespace-Cleanup 
Transient-Mark 
Undo-Tree 
Vi-Tilde-Fringe
Volatile-Highlights 
Which-Key 
Window-Numbering 
Winner Xterm-Mouse 
Yas 
Yas-Global

**Comment by deb0ch** [comment]
Do you have line numbers on ? Does it still happen the same without line numbers ?

**Comment by ParallaxVT** [comment]
@deb0ch I've always get numbers are deactivated by setting `dotspacemacs-line-numbers nil`

**Comment by ParallaxVT** [comment]
Is there a way through the spacemacs configuration to remove minor-modes from  a major-mode? In this case web-mode.
I thought that I could narrow the search by removing all the minor modes and then  applying one by one and see if one of them is causing the problem.

**Comment by nixmaniack** [comment]
Does this happen in normal stock emacs with just web-mode? You can try from spacemacs with `SPC q D` `web-mode` `RET`.

One starting point would be to `(setq web-mode-hook nil)` but it may be the case that it's derived from `prog-mode` then you'll have to disable its hooks as well. 
Also, there could be dependent modes in minor modes, global-minor-modes which might be getting triggered. So not really sure if that's the best way to go.

**Comment by dlukes** [comment]
@nixmaniack It happens in vanilla, freshly installed Spacemacs, but not in stock Emacs (`emacs -q`); see details from my experience with this issue below.

#### Description :octocat:

XML editing with nXML mode is very unresponsive, the interface freezes for long periods of time (seconds) typically when performing searches and trying to jump to the next match, but also when just navigating the XML file using arrow keys / `hjkl`. Reproduced on both Arch Linux and macOS with a vanilla install of Spacemacs. With stock Emacs (`emacs -q`), nXML isn't laggy at all on the same file.

#### Reproduction guide :beetle:
- Start Emacs (possibly a vanilla install of Spacemacs to ensure same environment)
- Open a medium-sized XML file -- you can [download](https://owncloud.korpus.cz/index.php/s/bi1NdQitZLLwxKa) the particular file which brought the issue to my attention
- Start a search: type `/`, then `fonet`
- Try to find the next match: type `n`

*Observed behaviour:* :eyes: :broken_heart:
The next match takes a long time to be found, Emacs becomes unresponsive. Subsequent freezes are typically experienced when trying to navigate around the file using arrow keys / `hjkl`.

*Expected behaviour:* :heart: :smile:
The next match should be found pretty much immediately, as in stock Emacs. It's fairly hard to diagnose what the problem is since there are no warnings or errors -- just some expensive computation running repeatedly and blocking the interface, I suppose :(

#### System Info :computer:
- OS: gnu/linux
- Emacs: 24.5.1
- Spacemacs: 0.200.9
- Spacemacs branch: master (rev. 8e1af145)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(helm emacs-lisp)
```

**Comment by Ps-spectre** [comment]
I have the same issue with a big xml files (in my case >20k lines, >800Kb) in vanilla spacemacs too. Like @dlukes 
Tried on Windows 7 / Windows 10 / Linux mint. (i7, 8gb ram, ssd and on old PC with Core2Duo, 4gb, hdd) 

**Reproduction guide**: the same as above @dlukes
(h, j, k, l movements, searching and helm-swoop (spc h h) and soon the interface freezes for long periods of time (ctrl + g can't help))

Checked on fresh Linux Mint, Core2Duo, 4gb ram, hdd.
#### System Info :computer:
- OS: gnu/linux
- Emacs: 25.2.2
- Spacemacs: 0.200.9
- Spacemacs branch: master (rev. 8e1af145)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(helm emacs-lisp)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GCONF GSETTINGS NOTIFY LIBSELINUX GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 MODULES

I've tried different fonts, turn on / off line numbers,  dotspacemacs-mode-line-unicode-symbols nil and other hacks / advices.

**Comment by Ps-spectre** [comment]
Whenever i try to edit huge xml file my spacemacs freezes. 
But i decided to turn on profiler and wait on my i7 laptop while my emacs was frozen. 
And after several minutes emacs unfroze. I got profiler-report 
```

- timer-event-handler                                            9027  91%
 - apply                                                         9027  91%
  - sp-show--pair-function                                       9012  91%
   - #<compiled 0x2763019>                                       9008  91%
    - sp-get-thing                                               9008  91%
     - sp-get-expression                                         9006  91%
      - sp-get-stringlike-or-textmode-expression                 9006  91%
       - sp-get-textmode-stringlike-expression                   9001  91%
        - sp--find-next-textmode-stringlike-delimiter               8517  86%
         - sp--search-backward-regexp                            8505  86%
            sp--looking-back                                        2   0%
         + sp-get-pair                                              5   0%
           sp--skip-match-p                                         2   0%
        + sp--get-allowed-stringlike-list                         455   4%
        + sp--textmode-stringlike-regexp                           25   0%
       + sp-get-stringlike-expression                               4   0%
     + sp-skip-forward-to-symbol                                    2   0%
   + sp--get-allowed-regexp                                         3   0%
     sp--looking-back                                               1   0%
  + which-key--update                                              11   0%
  + savehist-autosave                                               3   0%
+ command-execute                                                 709   7%


```
It seems that problem is **smartparens** (smartparens.el)

### Workaround:
Disable smartparens in current buffer helps me. (M-x show-smartparens-mode) 

Now I can edit huge xml files with spacemacs! I'm happy now! :) 
(there are no freezes for few minutes)

**Comment by dlukes** [comment]
@Ps-spectre Thanks for the investigation, `show-smartparens-mode` really seems to have been the culprit! I missed nXML sorely, it's so nice to have it back :)

In order not to have to turn off smartparens manually each time, I put `(show-smartparens-global-mode -1)` in my `dotspacemacs/user-config`. This prevents `show-smartparens-mode` from "infiltrating" major modes which didn't ask for it (like nXML).
